### PR TITLE
 Generate one-hot encoding for bead-type interactions in a CG system

### PR DESCRIPTION
**Pull request by mehradans92**
- Added a method to compute one-hot encoding for bead-type interactions given `mapped_pos_btype`, `mapped_nlist_btype` and  `number_of_unique_btypes` in a CG system.
- Added test for the method.

### CODE REVIEW

File: htf/test-py/test_utils.py [hunk 1]
```diff
@@ -235,6 +235,81 @@ def test_compute_nlist_cut(self):
...
+            mapped_pos_with_type = tf.concat(
+                [mapped_pos, system_bead_types], axis=1)
+            mapped_nlist = hoomd.htf.compute_nlist(
+                mapped_pos_with_type, r_cut, CG_NN, box_size, sorted=False, return_types=True)
+            break
+        pos_btype = tf.cast(mapped_pos_with_type[..., -1], dtype=tf.int32)
+        nlist_btype = tf.cast(mapped_nlist[..., -1], dtype=tf.int32)
+        ohe_beadtype_interactions = hoomd.htf.compute_ohe_bead_type_interactions(
+            pos_btype, nlist_btype, n_bead_types)
+        print('func [30,30]', ohe_beadtype_interactions[30, 10])
```
Comment (hgandhi2411) [inline htf/test-py/test_utils.py:299]: Shouldn't this be `func [30,10]`?

File: htf/test-py/test_utils.py [hunk 2]
```diff
@@ -235,6 +235,81 @@ def test_compute_nlist_cut(self):
...
+        print('func [130,16]', ohe_beadtype_interactions[130, 16])
+        n_btypes = 6
+        m, n = tf.math.minimum(pos_btype[..., tf.newaxis], nlist_btype), tf.math.maximum(
+            pos_btype[..., tf.newaxis], nlist_btype)
+        one_hot_indices = m*(2*n_btypes - m + 1)//2 + n - m
+        total_interactions = n_btypes * (n_btypes-1) // 2 + n_btypes
+        interactions = tf.one_hot(one_hot_indices, depth=total_interactions)
+        print('manual [30,30]', interactions[30, 10])
```
Comment (hgandhi2411) [inline htf/test-py/test_utils.py:307]: Same thing here

Comment (mehradans92) [reply htf/test-py/test_utils.py:307]: Yes. I was using that line for debugging

File: htf/test-py/test_utils.py [hunk 3]
```diff
@@ -235,6 +235,69 @@ def test_compute_nlist_cut(self):
...
+            u, [protein_FF.names], selection="resname PHE")
+        number_of_molecules = len(molecule_mapping_index)
+        system_bead_types = tf.cast(
+            tf.repeat(bead_types, number_of_molecules, axis=0), dtype=tf.float32)
+        cg_mapping = hoomd.htf.sparse_mapping([mapping_FF for _ in molecule_mapping_index],
+                                              molecule_mapping_index)
+        CG_NN = 64
+        r_cut = 25
+        # disable sorting
+        hoomd.context.current.sorter.disable()
```
Comment (mehradans92) [inline htf/test-py/test_utils.py:280]: Not sure why the tests were passing for me, @whitead  and @geemi725  before and not for @hgandhi2411 and Github automated checker. Got it to pass by adding `sorter.disable()` here and adding kwarg `sorted=True` in [https://github.com/ur-whitelab/hoomd-tf/pull/275/files#diff-e290eb7ad7c075af78c795ff0235ac40e11403259f20f8840eb122766bfaf901R291](here).

Comment (whitead) [reply htf/test-py/test_utils.py:280]: Perhaps you do not need these? Maybe you just need to assume when you do your final assertions that the arrays may not be in the same order. Maybe you could just sort the arrays before the assertion to fix that.

**Review by whitead — APPROVED** [review]

**Pull request merged**